processing/src/main/java/io/druid/query/aggregation/DoubleMaxAggregatorFactory.java

```diff
@@ -157,7 +161,9 @@ public String getName()
   @Override
   public List<String> requiredFields()
   {
-    return fieldName != null ? Collections.singletonList(fieldName) : Parser.findRequiredBindings(expression);
+    return fieldName != null
+           ? Collections.singletonList(fieldName)
+           : Parser.findRequiredBindings(Parser.parse(expression, macroTable));
```

Comment:
It's true, but would require some refactoring of AggregatorUtil and its callers. I think it'd be better to do it in a separate patch.